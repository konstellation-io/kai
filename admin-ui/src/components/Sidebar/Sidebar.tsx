import React, { useRef } from 'react';
import { History, Location } from 'history';

import NavBar from '../NavBar/NavBar';
import LeftArrowIcon from '@material-ui/icons/KeyboardBackspace';

import styles from './Sidebar.module.scss';

export type Tab = {
  label: string;
  route: string;
  Icon?: any;
};
type Props = {
  title: string;
  subtitle?: string;
  tabs: Tab[];
  history: History;
  location: Location;
  onChange?: Function;
  subheader?: any;
};

function Sidebar({
  title,
  tabs,
  history,
  location,
  subtitle = '',
  onChange = function(idx: number) {},
  subheader = undefined
}: Props) {
  const backLocation = useRef(location.state && location.state.prevLocation);

  function onBackButton() {
    history.push(backLocation.current);
  }

  const actualRoute = location.pathname;
  const activeTab = tabs.map(tab => tab.route).indexOf(actualRoute);
  return (
    <div className={styles.container}>
      <div className={styles.header}>
        <div className={styles.back} onClick={onBackButton}>
          <LeftArrowIcon style={{ fontSize: '1rem' }} />
          <span className={styles.title}>{title}</span>
        </div>
        <div className={styles.description}>{subtitle}</div>
      </div>
      {subheader}
      <div className={styles.navButtons}>
        <NavBar tabs={tabs} onChange={onChange} defaultActive={activeTab} />
      </div>
    </div>
  );
}

Sidebar.propTypes = {};

export default Sidebar;
