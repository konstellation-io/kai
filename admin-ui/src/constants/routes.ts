export const HOME = '/';
export const LOGIN = '/login';
export const VERIFY_EMAIL = '/verify_email';
export const MAGIC_LINK = '/signin/:token';
export const DASHBOARD = '/dashboard';

export const RUNTIME = `${DASHBOARD}/runtime/:runtimeId`;
export const RUNTIME_STATUS = `${DASHBOARD}/runtime/:runtimeId/status`;
export const RUNTIME_VERSION_STATUS = `${DASHBOARD}/runtime/:runtimeId/status/:versionId`;
export const RUNTIME_VERSIONS = `${DASHBOARD}/runtime/:runtimeId/versions`;
export const RUNTIME_VERSION_CONFIGURATION = `${DASHBOARD}/runtime/:runtimeId/configuration/:versionId`;

export const RUNTIME_PATHS = [
  RUNTIME_VERSION_STATUS,
  RUNTIME_VERSION_CONFIGURATION,
  RUNTIME,
  RUNTIME_STATUS,
  RUNTIME_VERSIONS
];

export const SETTINGS = '/settings';
export const SETTINGS_GENERAL = `${SETTINGS}/general`;
export const SETTINGS_SECURITY = `${SETTINGS}/security`;

export const AUDIT = '/audit';

export const NEW_RUNTIME = '/new-runtime';
export const NEW_VERSION = `${DASHBOARD}/runtime/:runtimeId/versions/new_version`;
