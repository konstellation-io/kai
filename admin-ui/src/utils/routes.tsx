import React from 'react';
import { Link } from 'react-router-dom';

export const buildRoute = {
  runtime: function(route: string, runtimeId: string = ''): string {
    return route.replace(':runtimeId', runtimeId);
  },
  version: function(
    route: string,
    runtimeId: string = '',
    versionId: string = ''
  ): string {
    return route
      .replace(':runtimeId', runtimeId)
      .replace(':versionId', versionId);
  }
};

type Props = {
  children: any;
  to: string;
  LinkType?: typeof Link;
  linkProps?: object;
  disabled?: boolean;
};
export function ConditionalLink({
  LinkType = Link,
  children,
  to = '',
  disabled = false,
  linkProps = {}
}: Props) {
  return !disabled ? (
    <LinkType to={to} {...linkProps}>
      {children}
    </LinkType>
  ) : (
    children
  );
}
