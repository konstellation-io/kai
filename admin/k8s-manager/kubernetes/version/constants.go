package version

const basePathKRT = "/krt-files"
const basePathKRTName = "krt-base-path"
const natsURL = "kre-nats:4222"
