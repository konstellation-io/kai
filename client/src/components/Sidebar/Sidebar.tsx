import React from 'react';
import { History, Location } from 'history';

import NavBar from '../NavBar/NavBar';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { IconProp } from '@fortawesome/fontawesome-svg-core';
import {ICON} from '../../icons';

import styles from './Sidebar.module.scss';


export type Tab = {
  label: string,
  route: string
  icon?: IconProp,
}
type Props = {
  title: string,
  tabs: Tab[];
  history: History;
  location: Location;
  onChange?: Function;
};

function Sidebar({
  title,
  tabs,
  history,
  location,
  onChange = function(idx:number) {},
}:Props) {
  function onBackButton() {
    history.goBack();
  }

  const actualRoute = location.pathname;
  const activeTab = tabs.map(tab => tab.route).indexOf(actualRoute);
  return (
    <div className={styles.container}>
      <div className={styles.header}>
        <div className={styles.back} onClick={onBackButton}>
         <FontAwesomeIcon icon={ICON.LEFT} size='xs' />
          { title }
        </div>
        <div className={styles.description}>
          Fusce vehicula dolor arcu, sit amet.
        </div>
      </div>
      <div>
        <NavBar
          tabs={tabs}
          onChange={onChange}
          defaultActive={activeTab}
        />
      </div>
    </div>
  );
}

Sidebar.propTypes = {
};

export default Sidebar;
