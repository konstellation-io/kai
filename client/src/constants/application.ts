export const COLORS = {
  BLUE: '#00A5CC',
  GRAY_DISABLED: '#1D1E23',
};

export const ENDPOINT = {
  SUBMIT_MAGIC_LINK: 'sumbit_magic_link',
};
