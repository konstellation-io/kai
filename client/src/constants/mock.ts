type Runtime = {
  id: string;
  status: string;
  title: string;
  info: {
    type: string;
    date: string;
  }[];
  disbled?: boolean;
};
const RUNTIME_A = {
  id: '564645987',
  status: '',
  title: 'Electricity system prediction',
  info: [
    {
      type: 'CREATED',
      date: '07/12/2019',
    },
    {
      type: 'UPDATED',
      date: '07/18/2019',
    },
  ],
};
const RUNTIME_B = {
  id: '564645987',
  status: '',
  title: 'Electricity system prediction',
  info: [
    {
      type: 'CREATED',
      date: '07/12/2019',
    },
    {
      type: 'UPDATED',
      date: '07/18/2019',
    },
  ],
  disabled: true,
};
export const RUNTIMES = [
  RUNTIME_A,
  {
    id: '523564366',
    status: 'Discovery',
    title: 'DAS Intrusion Detection',
    info: [
      {
        type: 'CREATED',
        date: '07/12/2019',
      },
      {
        type: 'UPDATED',
        date: '07/18/2019',
      },
    ],
  },
  RUNTIME_A,
  RUNTIME_A,
  RUNTIME_A,
  RUNTIME_B,
  RUNTIME_B,
  RUNTIME_A,
  {
    id: '0969969963',
    status: 'POC',
    title: 'Electricity system prediction',
    info: [
      {
        type: 'CREATED',
        date: '07/12/2019',
      },
      {
        type: 'UPDATED',
        date: '07/18/2019',
      },
    ],
  },
  RUNTIME_A,
  RUNTIME_B,
  RUNTIME_B,
  RUNTIME_B,
];

export function addToRuntimes(newRuntime: Runtime) {
  RUNTIMES.push(newRuntime);
}
