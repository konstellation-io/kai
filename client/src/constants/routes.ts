export const HOME = '/';
export const LOGIN = '/login';
export const VERIFY_EMAIL = '/verify_email';
export const MAGIC_LINK = '/magic_link/:token';
export const DASHBOARD = '/dashboard';

export const RUNTIME = `${DASHBOARD}/runtime/:runtimeId`;

export const SETTINGS = '/settings';
export const SETTINGS_GENERAL = `${SETTINGS}/general`;
export const SETTINGS_SECURITY = `${SETTINGS}/security`;
export const SETTINGS_AUDIT = `${SETTINGS}/audit`;

export const NEW_RUNTIME = '/new-runtime';
