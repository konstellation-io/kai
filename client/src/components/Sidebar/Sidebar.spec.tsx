import React from 'react';
import { render, fireEvent, cleanup, RenderResult } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import { Router } from 'react-router-dom'
import * as ROUTE from '../../constants/routes';
import Sidebar, { Tab } from './Sidebar';
import { createMemoryHistory, History } from 'history';
import '@testing-library/jest-dom/extend-expect';

afterEach(cleanup);

const tabs:Tab[] = [
  {
    label: 'GENERAL',
    icon: undefined,
    route: ROUTE.SETTINGS_GENERAL
  },
  {
    label: 'SECURITY',
    icon: undefined,
    route: ROUTE.SETTINGS_SECURITY
  },
  {
    label: 'AUDIT',
    icon: undefined,
    route: ROUTE.SETTINGS_AUDIT
  },
];

function generateComponent() {
  const history = createMemoryHistory();
  history.push(ROUTE.HOME);
  history.push(ROUTE.SETTINGS);

  const wrapper = render(
    <MemoryRouter>
      <Router history={history}>
        <Sidebar
          title='SIDEBAR'
          history={history}
          location={history.location}
          tabs={tabs}
        />
      </Router>
    </MemoryRouter>
  );

  return [wrapper, history];
}

it('Renders Sidebar without crashing', () => {
  const {container} = generateComponent()[0] as RenderResult;
  expect(container).toMatchSnapshot();
});

it('Shows navigation buttons', () => {
  const {getByText} = generateComponent()[0] as RenderResult;

  expect(getByText('GENERAL')).toBeInTheDocument();
  expect(getByText('SECURITY')).toBeInTheDocument();
  expect(getByText('AUDIT')).toBeInTheDocument();
});

it('Changes route on button click', () => {
  const [{getByText}, history] = generateComponent() as [RenderResult, History];

  fireEvent.click(getByText('SIDEBAR'));
  expect(history.location.pathname).toBe(ROUTE.HOME);
});
