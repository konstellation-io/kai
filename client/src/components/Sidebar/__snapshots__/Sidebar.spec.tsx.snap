// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Sidebar without crashing 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="header"
    >
      <div
        class="back"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          style="font-size: 1rem;"
          viewBox="0 0 24 24"
        >
          <path
            d="M21 11H6.83l3.58-3.59L9 6l-6 6 6 6 1.41-1.41L6.83 13H21z"
          />
        </svg>
        SIDEBAR
      </div>
      <div
        class="description"
      >
        Fusce vehicula dolor arcu, sit amet.
      </div>
    </div>
    <div>
      <div
        class="container"
      >
        <a
          data-testid="buttonLink"
          href="/settings/general"
        >
          <div
            class="btn transparent left label"
            style="height: 56px; line-height: 56px;"
          >
            <span>
              GENERAL
            </span>
          </div>
        </a>
        <a
          data-testid="buttonLink"
          href="/settings/security"
        >
          <div
            class="btn transparent left label"
            style="height: 56px; line-height: 56px;"
          >
            <span>
              SECURITY
            </span>
          </div>
        </a>
        <a
          data-testid="buttonLink"
          href="/settings/audit"
        >
          <div
            class="btn transparent left label"
            style="height: 56px; line-height: 56px;"
          >
            <span>
              AUDIT
            </span>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
